apps/studio/src/components/AppNavbar/PublishButton.tsx

```diff
@@ -0,0 +1,73 @@
+import { Skeleton, useToast } from "@chakra-ui/react"
+import { Button } from "@opengovsg/design-system-react"
+
+import { withSuspense } from "~/hocs/withSuspense"
+import { trpc } from "~/utils/trpc"
+
+interface PublishButtonProps {
+  pageId: number
+  siteId: number
+}
+
+const SuspendablePublishButton = ({
+  pageId,
+  siteId,
+}: PublishButtonProps): JSX.Element => {
+  const toast = useToast()
+  const utils = trpc.useUtils()
+
+  const [currPage] = trpc.page.readPage.useSuspenseQuery({ pageId, siteId })
+
+  const publishFailureMsg =
+    "Failed to publish page. Please contact Isomer support."
+  const publishSuccessMsg = "Page published successfully"
+
+  const { mutate, isLoading } = trpc.page.publishPage.useMutation({
+    onSuccess: async (data) => {
+      if (data.versionId) {
+        toast({
+          status: "success",
+          title: publishSuccessMsg,
+        })
+      } else {
+        console.error(data.error)
+        toast({
+          status: "error",
+          title: publishFailureMsg,
+        })
+      }
+      await utils.page.readPage.invalidate({ pageId, siteId })
+    },
+    onError: async () => {
+      toast({
+        status: "error",
+        title: publishFailureMsg,
+      })
+      await utils.page.readPage.invalidate({ pageId, siteId })
```

Comment:
The main purpose of publish to update our SOT first -> hence a version will be created and draftId will be set to null. At this point even if codebuild initiation fails, since there is no draft to publish, the button should be disabled. 

To ensure the button is disabled, we should refetch the state